integration-tests/src/test_loop/tests/epoch_sync.rs

```diff
@@ -88,26 +94,30 @@ fn test_epoch_sync_from_genesis() {
                 .chain_store
                 .store()
                 .clone(),
-            None,
-        ));
+        );
     }
-    stores.push((create_test_store(), None)); // new node starts empty.
 
     // Properly shut down the previous TestLoopEnv.
     // We must preserve the tempdir, since state dumps are stored there,
     // and are necessary for state sync to work on the new node.
     let tempdir = TestLoopEnv { test_loop, datas: node_datas, tempdir }
         .shutdown_and_drain_remaining_events(Duration::seconds(5));
 
-    tracing::info!("Starting new TestLoopEnv with new node");
+    TestNetworkSetup { tempdir, genesis, accounts, stores }
+}
 
-    let clients = accounts.iter().take(NUM_CLIENTS + 1).cloned().collect_vec();
+fn bootstrap_node_via_epoch_sync(setup: TestNetworkSetup, source_node: usize) -> TestNetworkSetup {
```

Comment:
nit: would destructuring `TestNetworkSetup` help with readability? We won't have to do `setup.{stuff}` very time.